filechanges/opt/musicbox/read_ini.sh

```diff
@@ -11,281 +11,277 @@
-    local INI_ALL_SECTION="${VARNAME_PREFIX}__ALL_SECTIONS"
-    local INI_NUMSECTIONS_VARNAME="${VARNAME_PREFIX}__NUMSECTIONS"
-    if [ "${CLEAN_ENV}" = 1 ] ;then
-	eval unset "\$${INI_ALL_VARNAME}"
-    fi
-    unset ${INI_ALL_VARNAME}
-    unset ${INI_ALL_SECTION}
-    unset ${INI_NUMSECTIONS_VARNAME}
-
-    if [ -z "$INI_FILE" ] ;then
-	cleanup_bash
-	return 0
-    fi
-    
-    if ! check_ini_file ;then
-	cleanup_bash
-	return 1
-    fi
-
-    # Sanitise BOOLEANS - interpret "0" as 0, anything else as 1
-    if [ "$BOOLEANS" != "0" ]
-    then
-	BOOLEANS=1
-    fi
-
-
-    # }}} END Options
-
-    # }}} END Deal with command line args
-
-    local LINE_NUM=0
-    local SECTIONS_NUM=0
-    local SECTION=""
-    
-    # IFS is used in "read" and we want to switch it within the loop
-    local IFS=$' \t\n'
-    local IFS_OLD="${IFS}"
-    
-    # we need some optional shell behavior (shopt) but want to restore
-    # current settings before returning
-    local SWITCH_SHOPT=""
-    pollute_bash
-    
-    while read -r line || [ -n "$line" ]
-    do
+	local LINE_NUM=0
+	local SECTIONS_NUM=0
+	local SECTION=""
+	
+	# IFS is used in "read" and we want to switch it within the loop
+	local IFS=$' \t\n'
+	local IFS_OLD="${IFS}"
+	
+	# we need some optional shell behavior (shopt) but want to restore
+	# current settings before returning
+	local SWITCH_SHOPT=""
+	pollute_bash
+	
+	while read -r line || [ -n "$line" ]
+	do
 #echo line = "$line"
 
-	((LINE_NUM++))
+		((LINE_NUM++))
 
-	# Skip blank lines and comments
-	if [ -z "$line" -o "${line:0:1}" = ";" -o "${line:0:1}" = "#" ]
-	then
-	    continue
-	fi
+		# Skip blank lines and comments
+		if [ -z "$line" -o "${line:0:1}" = ";" -o "${line:0:1}" = "#" ]
+		then
+			continue
+		fi
 
-	# Section marker?
-	if [[ "${line}" =~ ^\[[a-zA-Z0-9_]{1,}\]$ ]]
-	then
+		# Section marker?
+		if [[ "${line}" =~ ^\[.*\]$ ]]
+		then
+			# Set SECTION var to name of section (strip [ and ] from section marker)
+			SECTION="${line#[}"
+			SECTION="${SECTION%]}"
+			read SECTION <<<"${SECTION}"
+			if ! [[ "${line}" =~ ^[[:print:]]+$  ]] ;then
+				echo "Error: Invalid section:" >&2
+				echo " ${LINE_NUM}: '$line'" >&2
+				cleanup_bash
+				return 1
+			fi
+			((SECTIONS_NUM++))
+
+			continue
+		fi
 
-	    # Set SECTION var to name of section (strip [ and ] from section marker)
-	    SECTION="${line#[}"
-	    SECTION="${SECTION%]}"
-	    eval "${INI_ALL_SECTION}=\"\${${INI_ALL_SECTION}# } $SECTION\""
-	    ((SECTIONS_NUM++))
+		# Are we getting only a specific section? And are we currently in it?
+		if [ ! -z "$INI_SECTION" ]
+		then
+			if [ "$SECTION" != "$INI_SECTION" ]
+			then
+				continue
+			fi
+		fi
 
-	    continue
-	fi
+		# Valid var/value line? (check for variable name and then '=')
+		# HACK: Handling of local-sqlite's 'directories' variables. Bypass lines ending in 'directory:'
+		#       as if they are valid.
+		if ! [[ "${line}" =~ ^[[:print:]]+[[:space:]]*directory:|= ]]
```

Comment:
Yes, I was running it from setsound.sh. For some reason that I cannot explain the script wouldn't work when trying to call it directly (or at all on my Mint machine) so I just settled for that since it at least sort of worked. I hate bash. 

Let's at least remove that `directories` setting from the file anyway, I'm going to blindly assume it's the default value so we don't need to explicitly set it. And it's making a long file even longer.